### DISCUSSION TITLE
How to set default value of ItemProgress in Setup() ?

### DISCUSSION THREAD

**Discussion by Hoczer** [forntoh/LcdMenu#134] [Q&A] [answered]
Hello.
I need to save and read "power" variable from eeprom and pass it to menu ItemProgress as default value. I need to keep user settings when device is going power down.
Help me pls.

```
MAIN_MENU(
  ITEM_TOGGLE("Rozpalanie", "ON", "OFF", rozpalanieCallback),
  ITEM_STRING_LIST("Tryb", workMode, 4, workModeSelect),
  ITEM_SUBMENU("Praca reczna", manualMenu),
  ITEM_PROGRESS("Nadmuch",power, ventPowerCallBack),
  ITEM_COMMAND("EXIT", exitMenu));
```

**Comment by forntoh** [comment]
Check this thread 

https://github.com/forntoh/LcdMenu/discussions/67

This might also help

https://github.com/forntoh/LcdMenu/discussions/56#discussioncomment-5791075

**Reply by Hoczer** [reply]
Thanks for answer but i cant get this to work :/
This is the code:
```
MAIN_MENU(
  ITEM_TOGGLE("Rozpalanie", "ON", "OFF", rozpalanieCallback),
  ITEM_STRING_LIST("Tryb", workMode, 4, workModeSelect),
  ITEM_SUBMENU("Praca reczna", manualMenu),
  ITEM_PROGRESS("Nadmuch", power, ventPowerCallBack),
```
and in setup()

```
  static char buffor[10];
  itoa(20,buffor,10);

  menu[4] ->setValue(buffor);
  menu.resetMenu();
```
what im doing wrong ?

**Reply by forntoh** [reply]
@Hoczer I just noticed that I didn't implement this functionality for `ItemProgress`, I will do it when I have some time. I'm busy with other projects right now.

**Reply by matjanss** [reply]
Coincidentally, I was running into the same issue :)

I temporarily resolved it by adding...

`void setValue(char* value) { progress = atoi(value); }`

to ItemProgress.h

Now, this obviously only works in specific cases (which is fine for me) and, as we tend to say in dutch, doesn't win a beauty contest, but it works for my purposes.

**Reply by Hoczer** [reply]
I managed to get it working. I have edited ItemProgress.h file and added this function: 
```
/**
     * Set new value for progress .
     *
     * @param valueP The new input value.
     */
    void setValueP(uint16_t valueP) override { this->progress = valueP; }
```
Than in MenuItem.h i added this line: 
   ` virtual void setValueP(uint16_t valueP){};`
In setup () function: 
```
menu[4]->setValueP(100);
menu.update();
```
Works perfect :)

**Comment by forntoh** [comment, marked as answer]
`setProgress` was added in v3.4.0, see #150 

**Full Changelog**: https://github.com/forntoh/LcdMenu/compare/3.3.4...3.4.0
